lib/SPIRV/SPIRVReader.cpp

```diff
@@ -4830,7 +4831,7 @@ void SPIRVToLLVM::transSourceLanguage() {
   SPIRVWord Ver = 0;
   SourceLanguage Lang = BM->getSourceLanguage(&Ver);
   if (Lang != SourceLanguageUnknown && // Allow unknown for debug info test
-      Lang != SourceLanguageOpenCL_C && Lang != SourceLanguageOpenCL_CPP)
+      Lang != SourceLanguageOpenCL_C && Lang != SourceLanguageCPP_for_OpenCL)
```

Comment:
Hi, @svenvh , @MrSidims , can we preserve old SourceLanguageOpenCL_CPP metadata? Currently spirv reader drop SourceLanguageOpenCL_CPP for sycl source https://github.com/intel/llvm/blob/sycl/clang/lib/CodeGen/CodeGenModule.cpp#L1360,L1362 on spirv->LLVM IR.